x/ccv/provider/keeper/throttle.go

```diff
@@ -85,15 +92,24 @@ func (k Keeper) HandlePacketDataForChain(ctx sdktypes.Context, consumerChainID s
 ) {
 	// Get slash packet data and trailing vsc matured packet data, handle it all.
 	slashFound, slashData, vscMaturedData, seqNums := k.GetSlashAndTrailingData(ctx, consumerChainID)
+	seqNumsHandled := []uint64{}
 	if slashFound {
 		slashPacketHandler(ctx, consumerChainID, slashData)
+		// Slash packet should always be the first packet in the queue, so we can safely append the first seqNum.
+		seqNumsHandled = append(seqNumsHandled, seqNums[0])
 	}
-	for _, vscMData := range vscMaturedData {
+	for idx, vscMData := range vscMaturedData {
+		if k.VSCMaturedHandledLimitReached(ctx) {
+			// Break from for-loop, DeleteThrottledPacketData will still be called for this consumer
+			break
+		}
 		vscMaturedPacketHandler(ctx, consumerChainID, vscMData)
+		// Append seq num for this vsc matured packet
+		seqNumsHandled = append(seqNumsHandled, seqNums[idx+1]) // Note idx+1, since slash packet is at index 0
```

Comment:
I'm not a fan of assuming that `seqNums` will be constructed in s.t. the value at index 0 always corresponds to the slash packet, and all trailing indexes correspond to vsc matured. 

The tests around `GetSlashAndTrailingData` do validate the behavior I've described. However a better solution would be to refactor `GetSlashAndTrailingData` so that it returns structs that look something like..

```go
type slashPacketDataWithSeqNum struct {
	ibcSeqNum uint64
	data      ccvtypes.SlashPacketData
}

type vscMaturedPacketDataWithSeqNum struct {
	ibcSeqNum uint64
	data      ccvtypes.VSCMaturedPacketData
}
```

This is a trivial code change, but it'll likely make a large diff in `throttle_test.go`